### PR TITLE
fix: Consolidate logging for happy path.

### PR DESCRIPTION
**Pull request by Matovidlo**
Include ctxAttrs for logging properties of file.
Refactor telemetry to not be dependend on clk.

Jira: XXXXX

**Changes:**
- ...

-----------

### CODE REVIEW

File: internal/pkg/service/stream/storage/node/readernode/sliceupload/operator.go [hunk 1]
```diff
@@ -188,53 +213,44 @@ func (o *operator) uploadSlice(ctx context.Context, volume *diskreader.Volume, d
...
-			// Increment retry delay
-			err = o.storage.Slice().IncrementRetryAttempt(data.Slice.SliceKey, o.clock.Now(), err.Error()).Do(ctx).Err()
-			if err != nil {
-				o.logger.Errorf(ctx, "cannot increment slice retry: %v", err)
-				return
-			}
-
-			data.Processed = true
+			err = errors.PrefixErrorf(err, "cannot upload slice to staging")
+			// TODO: shouldnt be dbCtx used here?
```
Comment (Matovidlo) [inline internal/pkg/service/stream/storage/node/readernode/sliceupload/operator.go:222]: Ctx, cancel vs DbCtx, DbCancel, revision

File: internal/pkg/service/stream/storage/node/readernode/sliceupload/operator_test.go [hunk 2]
```diff
@@ -129,9 +129,12 @@ func TestSliceUpload(t *testing.T) {
...
-		logger.AssertJSONMessages(c, `{"level":"info","message":"uploading slice \"123/111/my-source/my-keboola-sink/2000-01-01T00:00:00.000Z/vol-1/2000-01-01T00:00:00.000Z\""}`)
-		logger.AssertJSONMessages(c, `{"level":"info","message":"slice was uploaded \"123/111/my-source/my-keboola-sink/2000-01-01T00:00:00.000Z/vol-1/2000-01-01T00:00:00.000Z\""}`)
-		logger.AssertJSONMessages(c, `{"level":"info","message":"slice was uploaded \"123/111/my-source/my-keboola-sink/2000-01-01T00:00:00.000Z/vol-2/2000-01-01T00:00:00.000Z\""}`)
+		logger.AssertJSONMessages(c, `{"level":"info","message":"upload slice","slice.key":"2000-01-01T00:00:00.000Z","slice.state":"uploading","component":"storage.node.operator.slice.upload"}`)
+		logger.AssertJSONMessages(c, `{"level":"info","message":"uploading slice to staging","slice.key":"2000-01-01T00:00:00.000Z","slice.state":"uploading","component":"storage.node.operator.slice.upload"}`)
+		// TODO: uploaded state?
+		// waitForSlicesSync(t)
+		logger.AssertJSONMessages(c, `{"level":"info","message":"successfully uploaded slice","slice.key":"2000-01-01T00:00:00.000Z","slice.state":"uploaded","component":"storage.node.operator.slice.upload"}`)
```
Comment (Matovidlo) [inline internal/pkg/service/stream/storage/node/readernode/sliceupload/operator_test.go:136]: Should here be `slice.state` `uploading` or `uploaded` as the model have changed but the logger still sees the slice in previous state?

File: internal/pkg/service/stream/storage/model/file.go [hunk 3]
```diff
@@ -75,16 +74,14 @@ func (f File) LastStateChange() utctime.UTCTime {
...
+func (f File) Telemetry() []attribute.KeyValue {
 	lastStateChange := f.LastStateChange().Time()
 	return []attribute.KeyValue{
 		attribute.String("project.id", f.ProjectID.String()),
 		attribute.String("branch.id", f.BranchID.String()),
 		attribute.String("source.id", f.SourceID.String()),
 		attribute.String("sink.id", f.SinkID.String()),
 		attribute.String("file.id", f.FileID.String()),
-		attribute.String("file.age", clk.Since(lastStateChange).String()),
-		attribute.String("file.state", f.State.String()),
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/model/file.go:87]: Because ctx attrs are set before operation and not updated, ... so old state was confusing in the output logs.

The `state` is useful in the `cleanup` package, because it determines the file expiration,
but in operators, we don't need it (or we have to manually update ctxattrs after a successful operation).

[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/model/file.go:87]: We had discussion here:
https://keboolaglobal.slack.com/archives/C055WKPHYSD/p1724743220728909?thread_ts=1724739535.521059&cid=C055WKPHYSD
TL;DR
the state is changed in all `operators` on `coordinator` or `readernode` level, but is read from mirror, which is older than current state of slice/file, therefore we do not want to propagate the `state`.

We had also discussion on `age` and the reason behind it is, that the `age` is important only during cleanup of  `file` but not important when changing file states and makes noise in logging. Therefore the `cleanup/node` adds this attributes and simplyfies dependendencies of `keboola.bridge` module.

File: internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go [hunk 4]
```diff
@@ -248,38 +256,62 @@ func (o *operator) rotateSlice(ctx context.Context, slice *sliceData) {
 
 	// Rotate slice
 	o.logger.Infof(ctx, "rotating slice for upload: %s", cause)
+	err = o.rotateSliceWithState(ctx, slice)
+	if err != nil {
+		return
+	}
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go:262]: I think you mean this code.
https://github.com/keboola/keboola-as-code/blob/fix-node-operator-logging-consolidation/internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go#L281-L290
If you can see, we can receive error when switching state to `close`.
In case we get error of this switching operation we try to increment the RetryAttempt. However, I use another error `iErr` to differentiate between these two.
1. In case we cannot increment attempt, it is  `!processed`
2. In case we can switch the increment attempt the function returns `nil` which set's it to `processed`.
3. In case we have no error is set's it to `processed`.

I've had different refactor in place but I decided to not go with it as your code was approved.

File: internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go [hunk 5]
```diff
@@ -248,38 +256,62 @@ func (o *operator) rotateSlice(ctx context.Context, slice *sliceData) {
...
+	o.logger.Infof(ctx, "closing slice")
+
 	// Wait until no source node is using the slice
 	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), o.config.SliceCloseTimeout.Duration())
 	defer cancel()
 
+	err := o.switchSliceToImporting(ctx, slice)
+	if err != nil {
+		return
+	}
```
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go:306]: I think you mean this code.
https://github.com/keboola/keboola-as-code/blob/fix-node-operator-logging-consolidation/internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go#L331-L341
        If you can see, we can receive error when switching state to `uploading`.
In case we get error of this switching operation we try to increment the RetryAttempt. However, I use another error `rErr` to differentiate between these two.
1. In case we cannot increment attempt, it is  `!processed`
2. In case we can switch the increment attempt the function returns `nil` which set's it to `processed`.
3. In case we have no error is set's it to `processed`.

I've had different refactor in place but I decided to not go with it as your code was approved.

[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go]: I changed `iErr` -> `rErr`, just cosmetic thing

**Comment by michaljurecko** [comment]
@jachym-tousek-keboola please, do another review on it, is it the final version?

File: internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go:262]: ping @Matovidlo

[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go:262]: should be fixed https://github.com/keboola/keboola-as-code/pull/1960/commits/454749fc6caec2df4d38fd6cc05a96390933d3d4

File: internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Matovidlo) [reply internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go]: should be fixed https://github.com/keboola/keboola-as-code/pull/1960/commits/454749fc6caec2df4d38fd6cc05a96390933d3d4

**Review by michaljurecko — APPROVED** [review]
